### PR TITLE
Issue 201 lock

### PR DESCRIPTION
**Pull request by cakirke**
address #201 - support additional File::NFSLock capabilities, stale lock expiry, errstr in thrown exceptions - will include lock holders in next version of File::NFSLock

### CODE REVIEW

File: t/02-bowels/80-repo-lock.t [hunk 1]
```diff
@@ -113,4 +114,29 @@ my $t = Pinto::Tester->new;
 }
 
 #------------------------------------------------------------------------------
+subtest 'Test stale lock file' => sub {
+
+    # create dummy lock file not connected to us
+    my $lockfile = $t->root->file('.lock');
+    my $fh = IO::File->new( $lockfile, 'w' );
```
Comment (thaljef) [inline t/02-bowels/80-repo-lock.t]: You could also just say `$lockfile->openw;` to open the file for writing (see the perldoc for `Path::Class::File`).

Or is there a particular reason for using `IO::File`? If so, then we also need to `use IO::File` in this test.

**Comment by thaljef** [comment]
I suppose stealing "stale" lockfiles is pretty common, so I'm ok with this. But it makes me a bit nervous. Personally, I would only want to steal a lockfile if I knew the current holder was dead and/or I gave permission to steal it (it seems that setting the stale timeout is tantamount to giving permission). Would it be prudent to issue a warning when this happens?

**Comment by thaljef** [comment]
Note: SQLite has its own locks, so that gives us a bit of a safety net.

File: t/02-bowels/80-repo-lock.t (hunk 1 above)
Comment (cakirke) [inline t/02-bowels/80-repo-lock.t]: no specific reason for IO::File, my first encounter with Path::Class::File and didn't dig deep enough, pushed the correction

**Comment by cakirke** [comment]
+1 on nervous, agree on setting environment = acknowledgement/permission but an appropriate message wouldn't hurt

**Comment by thaljef** [comment]
Hmm, I thought this was included in the 0.09998 release, but it appears that I never merged it.

**Pull request merged**